graphql_compiler/cost_estimation/cardinality_estimator.py

```diff
@@ -82,57 +82,120 @@ def _get_parent_and_child_name_from_edge(schema_graph, child_location):
     return parent_name_from_edge, child_name_from_edge
 
 
+def _get_outbound_and_inbound_names_from_locations(query_metadata, child_location, parent_location):
+    """Get the out and in classnames of a location and its parent from the last edge information."""
+    edge_direction, _ = _get_last_edge_direction_and_name_to_location(child_location)
+    parent_name_from_location = query_metadata.get_location_info(parent_location).type.name
+    child_name_from_location = query_metadata.get_location_info(child_location).type.name
+    if edge_direction == INBOUND_EDGE_DIRECTION:
+        outbound_vertex_name = child_name_from_location
+        inbound_vertex_name = parent_name_from_location
+    elif edge_direction == OUTBOUND_EDGE_DIRECTION:
+        outbound_vertex_name = parent_name_from_location
+        inbound_vertex_name = child_name_from_location
+    return outbound_vertex_name, inbound_vertex_name
+
+
+def _probe_statistics_for_edge_count_between_vertex_pair(
+    statistics, query_metadata, child_location, parent_location
+):
+    """Probe statistics for the count of edges connecting two vertex classes."""
+    _, edge_name = _get_last_edge_direction_and_name_to_location(child_location)
+    outbound_vertex_name, inbound_vertex_name = _get_outbound_and_inbound_names_from_locations(
+        query_metadata, child_location, parent_location
+    )
+    probe_result = statistics.get_edge_count_between_vertex_pair(
+        outbound_vertex_name, inbound_vertex_name, edge_name
+    )
+    return probe_result
+
+
+def _estimate_edge_count_between_vertex_pair_using_class_count(
+    schema_graph, statistics, query_metadata, child_location, parent_location
+):
+    """Estimate the count of edges connecting two vertex classes via the class_count statistic.
+
+    Given a parent location of type A and a child location of type B, we estimate the number of AB
+    edges using class counts. If A and B are subclasses of C and D respectively, we only have access
+    to CD edges, so in general, we'll use (# of CD edges) / (# of C vertices), but since we're only
```

Comment:
Shouldn't you multiply this by the number of A vertices too?

Reply:
Good catch!